WordPress/WordPressTest/WPAppAnalyticsTests.m

```diff
@@ -0,0 +1,164 @@
+//
+//  WPAppAnalyticsTests.m
+//  WordPress
+//
+//  Created by Diego E. Rey Mendez on 3/16/15.
+//  Copyright (c) 2015 WordPress. All rights reserved.
+//
+
+#import <OCMock/OCMock.h>
+#import <WordPressCom-Analytics-iOS/WPAnalytics.h>
+#import <XCTest/XCTest.h>
+
+#import "WordPressAppDelegate.h"
+#import "WordPressComApiCredentials.h"
+#import "WPAppAnalytics.h"
+#import "WPAnalyticsTrackerMixpanel.h"
+#import "WPAnalyticsTrackerWPCom.h"
+
+typedef void(^OCMockInvocationBlock)(NSInvocation* invocation);
+
+@interface WPAppAnalyticsTests : XCTestCase
+@end
+
+@implementation WPAppAnalyticsTests
+
+- (void)testInitializationWithMixPanelAndWPComTracker
+{
+    [[NSUserDefaults standardUserDefaults] setBool:YES forKey:WPAppAnalyticsDefaultsKeyUsageTracking];
+    
+    id analyticsMock = [OCMockObject mockForClass:[WPAnalytics class]];
+    id apiCredentialsMock = [OCMockObject mockForClass:[WordPressComApiCredentials class]];
+    
+    OCMockInvocationBlock firstRegisterTrackerInvocationBlock = ^(NSInvocation *invocation) {
+        __unsafe_unretained id<WPAnalyticsTracker> tracker = nil;
+        [invocation getArgument:&tracker atIndex:2];
+        
+        NSAssert([tracker isKindOfClass:[WPAnalyticsTrackerMixpanel class]],
+                 @"Expected to have a mixpanel tracker.");
+    };
+    
+    OCMockInvocationBlock secondRegisterTrackerInvocationBlock = ^(NSInvocation *invocation) {
+        __unsafe_unretained id<WPAnalyticsTracker> tracker = nil;
+        [invocation getArgument:&tracker atIndex:2];
+        
+        NSAssert([tracker isKindOfClass:[WPAnalyticsTrackerWPCom class]],
+                 @"Expected to have a WPCom tracker.");
+    };
+    
+    [[[apiCredentialsMock expect] andReturn:@"NON_EMPTY_TOKEN"] mixpanelAPIToken];
```

Comment:
I think the syntax works either way with the `andReturn` and the `mixpanelAPIToken` but wouldn't the reverse order read more natural, e.g. `[[[apiCredentialsMock expect] mixpanelAPIToken] andReturn:@"NON_EMPTY_TOKEN"]`